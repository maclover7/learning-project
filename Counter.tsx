import React, {Component} from 'react';
import {Button, StyleSheet, Text, View} from 'react-native';

const styles = StyleSheet.create({
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  button: {
    color: "#841584"
  }
});

interface IProps {}
interface IState {
  count: number;
}
export default class Counter extends Component<IProps, IState> {
  private state = { count: 0 }

  public render() {
    changeCount = (changeAmount: number) => {
      this.setState((oldState) => ({ count: oldState.count + changeAmount }));
    }

    increaseCount = () => { changeCount(1); };
    decreaseCount = () => { changeCount(-1); };

    return (
      <View>
        <Text style={styles.welcome}>Count: {this.state.count}</Text>
        <Button
          onPress={increaseCount}
          title="Increase Count"
          color={styles.button.color}
          accessibilityLabel="Increase Count"
        />
        <Button
          onPress={decreaseCount}
          title="Decrease Count"
          color={styles.button.color}
          accessibilityLabel="Decrease Count"
        />
      </View>
    );
  }
}
