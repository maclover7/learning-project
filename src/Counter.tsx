import React, { Component } from 'react';
import { Button, StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  button: {
    color: '#841584',
  },
});

interface IProps {}
interface IState {
  count: number;
}
export default class Counter extends Component<IProps, IState> {
  public state = { count: 0 };

  private increaseCount: () => void = () => {
    const changeAmount = 1;
    this.setState(oldState => ({ count: oldState.count + changeAmount }));
  };

  private decreaseCount: () => void = () => {
    const changeAmount = -1;
    this.setState(oldState => ({ count: oldState.count + changeAmount }));
  };

  public render() {
    return (
      <View>
        <Text style={styles.welcome}>Count: {this.state.count}</Text>
        <Button
          onPress={this.increaseCount}
          title={'Increase Count'}
          color={styles.button.color}
          accessibilityLabel={'Increase Count'}
        />
        <Button
          onPress={this.decreaseCount}
          title={'Decrease Count'}
          color={styles.button.color}
          accessibilityLabel={'Decrease Count'}
        />
      </View>
    );
  }
}
