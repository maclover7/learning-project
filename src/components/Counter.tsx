import React from 'react';
import { Button, StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  button: {
    color: '#841584',
  },
});

interface IProps {
  count: number;
  decreaseCount: () => void;
  increaseCount: () => void;
}

const Counter = (props: IProps) => {
  return (
    <View>
      <Text style={styles.welcome}>Count: {props.count}</Text>
      <Button
        onPress={props.increaseCount}
        title={'Increase Count'}
        color={styles.button.color}
        accessibilityLabel={'Increase Count'}
      />
      <Button
        onPress={props.decreaseCount}
        title={'Decrease Count'}
        color={styles.button.color}
        accessibilityLabel={'Decrease Count'}
      />
    </View>
  );
};

export default Counter;
