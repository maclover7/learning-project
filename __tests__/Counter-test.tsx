import React from 'react';
import Counter from '../src/components/Counter';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const component = renderer.create(<Counter count={0}
                                             decreaseCount={() => {}}
                                             increaseCount={() => {}} />);
  expect(component.toJSON()).toMatchSnapshot();
});
