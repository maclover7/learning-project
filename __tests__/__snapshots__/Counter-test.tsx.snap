// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Count: 
    0
  </Text>
  <View
    accessibilityLabel="Increase Count"
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#841584",
            },
          ]
        }
      >
        Increase Count
      </Text>
    </View>
  </View>
  <View
    accessibilityLabel="Decrease Count"
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#841584",
            },
          ]
        }
      >
        Decrease Count
      </Text>
    </View>
  </View>
</View>
`;
